skyportal/handlers/api/telescope.py

```diff
@@ -66,32 +66,57 @@ def post(self):
         return self.success(data={"id": telescope.id})
 
     @auth_or_token
-    def get(self, telescope_id):
+    def get(self, telescope_id=None):
         """
         ---
-        description: Retrieve a telescope
-        parameters:
-          - in: path
-            name: telescope_id
-            required: true
-            schema:
-              type: integer
-        responses:
-          200:
-            content:
-              application/json:
-                schema: SingleTelescope
-          400:
-            content:
-              application/json:
-                schema: Error
+        single:
+          description: Retrieve a telescope
+          parameters:
+            - in: path
+              name: telescope_id
+              required: true
+              schema:
+                type: integer
+          responses:
+            200:
+              content:
+                application/json:
+                  schema: SingleTelescope
+            400:
+              content:
+                application/json:
+                  schema: Error
+        multiple:
+          description: Retrieve all telescopes
+          parameters:
+            - in: query
+              name: name
+              schema:
+                type: string
+              description: Filter by name (exact match)
+          responses:
+            200:
+              content:
+                application/json:
+                  schema: ArrayOfTelescopes
+            400:
+              content:
+                application/json:
+                  schema: Error
         """
-        t = Telescope.get_if_owned_by(int(telescope_id), self.current_user)
-
-        if t is not None:
+        if telescope_id is not None:
+            t = Telescope.get_if_owned_by(int(telescope_id), self.current_user)
+            if t is None:
+                return self.error(f"Could not load telescope with ID {telescope_id}")
             return self.success(data=t)
-        else:
-            return self.error(f"Could not load telescope with ID {telescope_id}")
+        tel_name = self.get_query_argument("name", None)
+        query = Telescope.query.filter(Telescope.id.in_(
```

Comment:
Maybe we should expand `get_if_owned_by` to take multiple IDs in another PR.  Thoughts?